src/fitnesse/reporting/history/PageHistoryReader.java

```diff
@@ -29,19 +29,24 @@ private boolean fileIsNotADirectoryAndIsValid(File file) {
   public static boolean matchesPageHistoryFileFormat(String pageHistoryFileName) {
     return pageHistoryFileName.matches(TEST_FILE_FORMAT);
   }
-  
-  private void compileResultFileIntoHistory(File file) throws ParseException {
+
+  private void compileResultFileIntoHistory(File file) {
     TestResultRecord record = buildTestResultRecord(file);
     processTestFile(record);
   }
 
-  void processTestFile(TestResultRecord record) throws ParseException {
+  void processTestFile(TestResultRecord record) {
     // for subclasses.
   }
-  
-  private TestResultRecord buildTestResultRecord(File file) throws ParseException {
+
+  private TestResultRecord buildTestResultRecord(File file) {
     String[] parts = file.getName().split("_|\\.");
-    Date date = dateFormat.parse(parts[0]);
+    Date date;
+    try {
+      date = dateFormat.parse(parts[0]);
+    } catch (ParseException e) {
+      throw new IllegalStateException("File format should have been verified", e);
```

Comment:
"File format should have been verified" does not seem like the most descriptive message when date parsing fails....